tests/ui/missing_const_for_fn/could_be_const.rs

```diff
@@ -65,7 +65,7 @@ mod with_drop {
 
     impl B {
         // This can be const, because `a` is passed by reference
-        pub fn b(self, a: &A) -> B {
+        pub fn new(a: &A) -> B {
```

Comment:
Not sure if that's important, but why the `self` arg was removed? I don't think that semantics of other tests should be affected.

Reply:
If the `self` isn't removed, then `clippy::wrong_self_convention` gets thrown because the method `b` became `new`.

Reply:
I mean, you can leave it as it was. `fn b(self, a: &A)` should not throw either `wrong_self_convention` or `self_named_constructor`.

Reply:
It seems to be throwing `self_named_constructor`. Should I add a check for the existence of `self`, which would imply that it is not a constructor?

Reply:
Ah, sure! If it has `self`, it's surely not a constructor. For some reason, I supposed that this check exists already.